lib/ibmdb.js

```diff
@@ -255,24 +255,16 @@ IBMDB.prototype.executeSQL = function(sql, params, options, callback) {
     stmt.sql = sql;
     stmt.params = params;
 
-    conn.query(stmt, function(err, data, more) {
+    conn.query(stmt, function(err, data, sqlca) {
       debug('IBMDB.prototype.executeSQL (exit)' +
-      ' stmt=%j params=%j err=%j data=%j more=%j',
-      stmt, params, err, data, more);
-      // schedule callback as there is more code to
-      // execute in the db2 driver to cleanup the current query
+      ' stmt=%j params=%j err=%j data=%j sqlca=%j',
+      stmt, params, err, data, sqlca);
+
+      // FIXME: A better way for pagination
       if (offset || limit) {
         data = data.slice(offset, offset + limit);
       }
 
-      if (!err) {
```

Comment:
Not sure what's the original purpose to return the callback in next tick when `!err && more == true`, to keep it we can do this:

```js
if (!err && more) {
  process.nextTick(function() {return cb(err, data)})
} else {
  return cb && cb(err, data);
}
```

thought?

Reply:
I checked the documentation - https://github.com/ibmdb/node-ibm_db/blob/master/APIDocumentation.md#queryApi. The 3rd arg of the callback is `sqlca`, not `more`.